### PR TITLE
DD-648 fix for allow custom terms

### PR DESCRIPTION
**Pull request by mderuijter**
**What this PR does / why we need it**:
Only sets a custom license when :AllowCustomLicense is set to true, otherwise default license is used
**Which issue(s) this PR closes**:

Closes [#DD-648](https://drivenbydata.atlassian.net/browse/DD-648)

For testing see issue

### CODE REVIEW

File: src/main/java/edu/harvard/iq/dataverse/util/json/JsonParser.java [hunk 1]
```diff
@@ -1,23 +1,7 @@
 package edu.harvard.iq.dataverse.util.json;
 
 import com.google.gson.Gson;
-import edu.harvard.iq.dataverse.ControlledVocabularyValue;
```
Comment (janvanmansum) [inline src/main/java/edu/harvard/iq/dataverse/util/json/JsonParser.java]: @mderuijter is your IDE set to replace imports with `*` imports? If so, please turn that off.

Comment (mderuijter) [reply src/main/java/edu/harvard/iq/dataverse/util/json/JsonParser.java]: @janvanmansum I think so... Let me check

File: src/main/java/edu/harvard/iq/dataverse/util/json/JsonParser.java [hunk 2]
```diff
@@ -360,15 +376,20 @@ public DatasetVersion parseDatasetVersion(JsonObject obj, DatasetVersion dsv) th
...
             }
-            return null;
-        } catch (URISyntaxException e) {
-            return null;
-        }       
+        }
+        try {
+            return licenseService.getByNameOrUri(licenseNameOrUri);
+        } catch (NoResultException nre){
+            throw new JsonParseException("Couldn't find an active license with: " + licenseNameOrUri);
```
Comment (janvanmansum) [inline src/main/java/edu/harvard/iq/dataverse/util/json/JsonParser.java]: @mderuijter : is really a JSON parsing problem??

Comment (mderuijter) [reply src/main/java/edu/harvard/iq/dataverse/util/json/JsonParser.java]: @janvanmansum no, I see now that it's a silly mistake. My train of thought was that only JsonParseExceptions are used here and I had to use that for catching the NoResultException, I realize now it is wrong. 

I need to readjust that anyway according to the other exception refactoring

Comment (mderuijter) [reply src/main/java/edu/harvard/iq/dataverse/util/json/JsonParser.java]: @janvanmansum had another look at the code in this class and the JsonParseException is actually used for every error that happens during the parsing process: [JsonParser.java#L575-L582](https://github.com/DANS-KNAW/dataverse/blob/8c8c93ba986d39e79d1bd6d605d8781188673df2/src/main/java/edu/harvard/iq/dataverse/util/json/JsonParser.java#L575-L582). 
So I will change my code and remove the NoResultException try-catch and return a JsonParseException if the License == null

**Pull request merged**